src/journal-remote/journal-remote-main.c

```diff
@@ -203,6 +205,8 @@ static int process_http_upload(
         bool finished = false;
         size_t remaining;
         int r;
+        char *buf = NULL;
+        size_t buf_size;
```

Comment:
Please move the declarations of `buf` and `buf_size` to `if (*upload_data_size)` branch below.

Also, `buf` seems to be leaked. Please use `_cleanup_free_` attribute.